addons/mail/models/mail_message.py

```diff
@@ -936,7 +952,22 @@ def _message_format(self, fnames, format_reply=True):
                 'record_name': record_name,
                 'create_date': message_sudo.create_date,
                 'write_date': message_sudo.write_date,
+                'needaction_partner_ids': notifs.filtered(lambda n: not n.is_read).res_partner_id.ids,
+                'starredPersonas': [{'id': id, 'type': "partner"} for id in message_sudo.starred_partner_ids.ids],
+                'history_partner_ids': notifs.filtered(lambda n: n.is_read).res_partner_id.ids,
+                'is_note': message_sudo.subtype_id.id == note_id,
+                'is_discussion': message_sudo.subtype_id.id == com_id,
+                'subtype_description': message_sudo.subtype_id.description,
+                'recipients': [{'id': p.id, 'name': p.name, 'type': "partner"} for p in message_sudo.partner_ids],
+                'scheduledDatetime': scheduled_dt_by_msg_id.get(vals['id'], False),
             })
+            if vals['model'] and vals['res_id']:
+                thread = {'model': vals['model'], 'id': vals['res_id']}
```

Comment:
as we're having `message_sudo` in this loop, it would be more clear to use it rather than reading values from `vals`